systemFlake.nix

```
@@ -26,24 +26,28 @@
 
 let
   evalHostArgs =
-    { channelName ? "nixpkgs"
+    { channelName ? null
     , modules ? []
-    , system ? defaultSystem
+    , system ? null
     , extraArgs ? {}
-    , ...
-    }: defaultHostAttrs
-      // { 
-        inherit channelName system; 
-        modules = sharedModules ++ modules;
-        extraArgs = sharedExtraArgs // extraArgs;
-      };
+    }: {
+      # hard coded fallbacks
+      channelName = "nixpkgs";
+      system = defaultSystem or "x86_64-linux";
```

```suggestion
    }: {
      # hard coded fallbacks
      inherit channelName system;
```
How about do that instead, so you can still set the defaults up top, dropping the need for `null` and still allowing `defaultHostAttrs` to get priority over some things.